app/models/payment_gateway_setting.rb

```diff
@@ -37,4 +39,11 @@ def active_gateway_type
       errors.add(:gateway_type, :invalid)
     end
   end
+
+  private
+
+  def publish_events
+    Accounts::PaymentSettingDeletedEvent.create_and_publish!(self)
+    true # object has to be destroyed, no matter if the event has been published
```

Comment:
So we can just publish_events after commit no?

Reply:
right 👍